pkg/messaging/mqtt/pubsub.go

```diff
@@ -80,7 +80,23 @@ func (ps pubsub) Subscribe(id, topic string, handler messaging.MessageHandler) e
 	case true:
 		// Check topic
 		if ok = s.contains(topic); ok {
-			return errAlreadySubscribed
+			// Unlocking, so that Unsubscribe() can access ps.subscriptions
+			ps.mu.Unlock()
+			if err := ps.Unsubscribe(id, topic); err != nil {
+				ps.mu.Lock() // Lock so that deferred unlock handle it
+				return err
+			}
+			ps.mu.Lock()
```

Comment:
```suggestion
			err := ps.Unsubscribe(id, topic)
			ps.mu.Lock() // Lock so that deferred unlock handle it
			if err != nil {
				return err
			}
```